### PR TITLE
Use HashWithIndifferentAccess for storing HMCTS raw data

### PR DESCRIPTION
**Pull request by laurentqro**
## What

Use HashWithIndifferentAccess for storing HMCTS raw data, so as to avoid not being able to access the hash key's value because we're accessing with symbolised rather than stringified keys.

Documentation: https://api.rubyonrails.org/classes/ActiveSupport/HashWithIndifferentAccess.html

We also split `hearing/all_fields.json` JSON fixture file into `hearing_resulted.json` and removed the `"sharedTime"` and `"hearing"` keys from `hearing/all_fields.json`, which are two distinct entities as per the schema, and are not be confused with one another.

### CODE REVIEW

File: app/models/hmcts_common_platform/court_application.rb [hunk 1]
```diff
@@ -2,12 +2,12 @@ module HmctsCommonPlatform
   class CourtApplication
     attr_reader :data
 
+    delegate :blank?, to: :data
+
     def initialize(data)
       @data = data || {}
```
Comment (arthurashman) [inline app/models/hmcts_common_platform/court_application.rb:8]: Any reason this isn't `@data = HashWithIndifferentAccess.new(data || {})` ?

File: app/models/hmcts_common_platform/delegated_powers.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@ class DelegatedPowers
     attr_reader :data
 
     def initialize(data)
-      @data = data || {}
+      @data = HashWithIndifferentAccess.new(data || {})
     end
 
     delegate :blank?, to: :data
```
Comment (arthurashman) [inline app/models/hmcts_common_platform/delegated_powers.rb:9]: Move this above initialize?

File: app/models/hmcts_common_platform/offence.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@ class Offence
     attr_reader :data
 
     def initialize(data)
-      @data = data || {}
+      @data = HashWithIndifferentAccess.new(data || {})
     end
 
     delegate :blank?, to: :data
```
Comment (arthurashman) [inline app/models/hmcts_common_platform/offence.rb:9]: Move this above initialize?

File: app/models/hmcts_common_platform/person_defendant.rb [hunk 4]
```diff
@@ -3,7 +3,7 @@ class PersonDefendant
     attr_reader :data
 
     def initialize(data)
-      @data = data || {}
+      @data = HashWithIndifferentAccess.new(data || {})
     end
 
     delegate :blank?, to: :data
```
Comment (arthurashman) [inline app/models/hmcts_common_platform/person_defendant.rb:9]: Move this above initialize?

File: app/models/hmcts_common_platform/plea.rb [hunk 5]
```diff
@@ -3,7 +3,7 @@ class Plea
     attr_reader :data
 
     def initialize(data)
-      @data = data || {}
+      @data = HashWithIndifferentAccess.new(data || {})
     end
 
     delegate :blank?, to: :data
```
Comment (arthurashman) [inline app/models/hmcts_common_platform/plea.rb:9]: Move this above initialize?

File: spec/models/hmcts_common_platform/defendant_spec.rb [hunk 6]
```diff
@@ -2,7 +2,7 @@
   let(:defendant) { described_class.new(data) }
 
   context "when the defendant has all fields" do
-    let(:data) { JSON.parse(file_fixture("defendant/all_fields.json").read).deep_symbolize_keys }
+    let(:data) { JSON.parse(file_fixture("defendant/all_fields.json").read) }
```
Comment (arthurashman) [inline spec/models/hmcts_common_platform/defendant_spec.rb:5]: Out of curiosity, do the serialisers in the app sybolize?

File: app/models/hmcts_common_platform/court_application.rb (hunk 1 above)
Comment (laurentqro) [reply app/models/hmcts_common_platform/court_application.rb]: there is a bad reason: I forgot this one :)

File: app/models/hmcts_common_platform/delegated_powers.rb (hunk 2 above)
Comment (arthurashman) [reply app/models/hmcts_common_platform/delegated_powers.rb:9]: Last one 😉

**Review by arthurashman — APPROVED** [review]

**Pull request merged**